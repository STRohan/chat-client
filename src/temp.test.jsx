import React from 'react'
import { shallow, mount } from 'enzyme'
import Temp from './temp';

/**
 * @description: Axis Label unit testing
 */
// const wrapper = shallow(<Temp />)

const findByTestAttr = (wrapper, val) => wrapper.find(`[data-test="${val}"]`);

describe('testing temp Component', () => {

  it('matches the snapshot', () => {
    expect(shallow(<Temp />)).toMatchSnapshot()
  })
})

it('should render component correctly', () => {
  const wrapper = shallow(<Temp />)
  const tempComponent = findByTestAttr(wrapper, 'component-text')
  expect(tempComponent.length).toBe(1)
})

// it('keep only onchange func', () => {
//   const onChangeMock = jest.fn();
//   const event = {
//     preventDefault() {},
//     target: { value: 'jhon whick' },
//   };
//   DateInputComponent = mount(<Temp {...props} />).find('value');

//   const  wrapper = shallow(<Temp onChange={onChangeMock} />)
//   wrapper.find('TextField').at('0').simulate('change', event);
//   expect(onChangeMock).toBeTruthy();
// })


it('keep only onchange func', () => {
  const onChangeMock = jest.fn();
  const event = {
    preventDefault() {},
    target: { value: 'heisenberg' },
  };
  const  wrapper = shallow(<Temp onChange={onChangeMock} />)
  wrapper.find('TextField').at('0').simulate('change', event);
  expect(wrapper.find('TextField').props().value).toEqual('heisenberg');
})

//-------------------------------------------------------------------------------------------
// it('check the onChange callback', () => {  
//   const onChange = jest.fn(),
//       props = {
//           value: 'rohan',
//           onChange
//       },
//       DateInputComponent = mount(<Temp {...props} />).find('value');
//   DateInputComponent.simulate('change', { target: {value: 'jhon wick'} });
//   expect(onChange).toHaveBeenCalledWith('jhonwick');
// });