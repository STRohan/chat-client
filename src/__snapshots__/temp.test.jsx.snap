// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing temp Component matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Temp />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <TextField
          data-test="component-text"
          id="outlined-name"
          label="Name"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          value="Cat in the Hat"
          variant="outlined"
        />,
        <div>
          Cat in the Hat
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data-test": "component-text",
          "id": "outlined-name",
          "label": "Name",
          "margin": "normal",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "Cat in the Hat",
          "variant": "outlined",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cat in the Hat",
        },
        "ref": null,
        "rendered": "Cat in the Hat",
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <TextField
            data-test="component-text"
            id="outlined-name"
            label="Name"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            value="Cat in the Hat"
            variant="outlined"
          />,
          <div>
            Cat in the Hat
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data-test": "component-text",
            "id": "outlined-name",
            "label": "Name",
            "margin": "normal",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "Cat in the Hat",
            "variant": "outlined",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cat in the Hat",
          },
          "ref": null,
          "rendered": "Cat in the Hat",
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
